exasol_integration_test_docker_environment/test/test_test_env_reuse.py

```diff
@@ -0,0 +1,142 @@
+                                    docker_db_image_version=self.docker_db_version_parameter,
+                                    docker_db_image_name="exasol/docker-db"
+                                    )
+        try:
+            success = luigi.build([task], workers=1, local_scheduler=True, log_level="INFO")
+            if success:
+                result = task
+            else:
+                task.cleanup(False)
+                Exception("Task failed")
+        except Exception as e:
+            task.cleanup(False)
+        return result
+
+    def _create_exaplus_check_cmd(self, test_environment_info):
+        username = SpawnTestEnvironment.DEFAULT_DB_USER
+        password = SpawnTestEnvironment.DEFAULT_DATABASE_PASSWORD
+        database_host = test_environment_info.database_info.host
+        database_port = test_environment_info.database_info.db_port
+        q = "SELECT TABLE_NAME FROM SYS.EXA_ALL_TABLES WHERE TABLE_SCHEMA='TEST';"
+        return f"""$EXAPLUS -c '{database_host}:{database_port}' -u '{username}' -p '{password}' -sql \\\"{q}\\\""""
+
+    def _exec_cmd_in_test_container(self, test_environment_info, cmd):
+        bash_cmd = f"""bash -c "{cmd}" """
+        test_container = self._client.containers.get(test_environment_info.test_container_info.container_name)
+
+        exit_code, output = test_container.exec_run(cmd=bash_cmd)
+        self.assertEquals(exit_code, 0)
+        return output.decode('utf-8')
+
+    def _verify_test_data(self, test_environment_info):
+        cmd = self._create_exaplus_check_cmd(test_environment_info)
+        output_str = self._exec_cmd_in_test_container(test_environment_info, cmd)
+
+        # TODO read /tests/test/import.sql and apply a regular expression to
+        # get all table names and compare then one-by-one
+        self.assertIn("ENGINETABLE", [output_entry.strip() for output_entry in output_str.split(sep='\n')])
+
+    def test_initial_reuse_database_setup_populates_data(self):
+        task = process_spawn_utils.run_in_process(self.run_spawn_test_env_with_cleanup)
+        self._verify_test_data(task.get_return_object())
+        task.cleanup(True)
+
+    def get_instance_ids(self, test_environment_info):
+        test_container = self._client.containers.get(test_environment_info.test_container_info.container_name)
+        db_container = self._client.containers.get(test_environment_info.database_info.container_info.container_name)
+        network = self._client.networks.get(test_environment_info.network_info.network_name)
+        return test_container.id, db_container.id, network.id
+
+    def test_reuse_env_same_instances(self):
+        task = process_spawn_utils.run_in_process(self.run_spawn_test_env_without_cleanup)
+        test_environment_info = task.get_return_object()
+        old_instance_ids = self.get_instance_ids(test_environment_info)
+        # This clean is supposed to not remove docker instances
+        task.cleanup(True)
```

Comment:
Because, the deserialized task probably has the wrong job id, its cleanup might not work as expected. But as discussed we can't fix this with the current way how we set the job id.